lib/baremetalbasetest.pm

```diff
@@ -0,0 +1,269 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Base module for baremetal test cases
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+package baremetalbasetest;
+
+use base opensusebasetest;
+
+use testapi;
+use strict;
+use warnings;
+
+use HTTP::Tiny;
+use IPC::Run;
+use Socket;
+use Time::HiRes 'sleep';
+
+
+# base class for baremetal tests
+sub new {
+    my ($class, $args) = @_;
+    my $self = $class->SUPER::new($args);
+
+    token = get_var('BAREMETAL_LOCK_TOKEN', '');
+    if (token == '') {
+        $self->host_lock();
+    }
+    return $self;
+}
+
+
+=head2 post_run_hook
+
+  post_run_hook();
+
+This method will be called after each module finished.
+It will B<not> get executed when the test module failed.
+Test modules (or their intermediate base classes) may overwrite
+this method.
+
+=cut
+sub post_run_hook {
+    my ($self) = @_;
+
+    die "only IPMI is supported for baremetal testing right now" unless check_var('BACKEND', 'ipmi');
+
+    $self->SUPER::post_run_hook();
+}
+
+=head2 post_fail_hook
+
+ post_fail_hook();
+
+When the test module fails, this method will be called.
+It will try to fetch some logs from the SUT.
+Test modules (or their intermediate base classes) may overwrite
+this method to export certain specific logfiles and call the
+base method using C<$self-E<gt>SUPER::post_fail_hook;> at the end.
+
+=cut
+sub post_fail_hook {
+    my ($self) = @_;
+
+    # usually, we are on a serial terminal, but we could of course
+    # also run on a VNC session, so support this by running the
+    # post fail hook from the base class
+    $self->SUPER::post_fail_hook();
+
+    # unlock the host if it was locked. Make sure not to keep the
+    # host locked for longer that needed.
+    if (host_is_locked()) {
+        host_unlock();
+    }
+
+}
+
+=head2 poweron
+
+ poweron()
+
+Power on the machine if the backend in use is supported. 
+=cut
+sub poweron {
+    if (check_var('BACKEND', 'ipmi')) {
+        ipmitool("chassis power on");
+        while (1) {
+            sleep(3);
```

Comment:
I'd rather avoid polluting the logs with that, as this will always loop for a few iterations, as the change in ipmitool's output takes a while. This is just to avoid a single lost packet will fail to power on the machine. I would prefer not polluting our logs with useless information.